src/clj/witan/send/model/data_products.clj

```diff
@@ -0,0 +1,193 @@
+     ([totals [need-setting population]]
+      (update totals need-setting m/some+ population)))
+   {}
+   calendar-year))
+
+(defn roll-up-calendar-year-by-state [all-valid-states calendar-year]
+  (reduce
+   (fn [acc state]
+     (assoc acc state (get calendar-year state 0)))
+   {}
+   all-valid-states))
+
+(defn roll-up-calendar-year-by-ay [calendar-year]
+  (reduce
+   (fn [acc [[ay need-setting] pop]]
+     (update acc ay + pop)) ;; can be + b/c we shouldn't have pops to add outside of the known academic years
+   (into (sorted-map) (map vector sc/academic-years (repeat 0)))
+   calendar-year))
+
+(defn roll-up-calendar-year-by-total-send-population [calendar-year]
+  (transduce
+   (remove (fn [[[ay need-setting] population]]
+             (= need-setting constants/non-send)))
+   (fn
+     ([total] total)
+     ([total [[ay need-setting] population]] (update total :total-population + population)))
+   {:total-population 0}
+   calendar-year))
+
+(defn roll-up-calendar-year-by-total-in-send-by-need [calendar-year]
+  (transduce
+   (map (fn [[[ay need-setting] population]]
+          [(-> need-setting
+               (states/split-need-setting)
+               first)
+           population]))
+   (fn
+     ([totals] totals)
+     ([totals [need population]]
+      (update totals need m/some+ population)))
+   {}
+   calendar-year))
+
+(defn roll-up-calendar-year-by-total-in-send-by-setting [calendar-year]
+  (transduce
+   (map (fn [[[ay need-setting] population]]
+          [(-> need-setting
+               (states/split-need-setting)
+               second)
+           population]))
+   (fn
+     ([totals] totals)
+     ([totals [setting population]]
+      (update totals setting m/some+ population)))
+   {}
+   calendar-year))
+
+(defn roll-up-calendar-year-by-total-cost [cost-lookup calendar-year]
+  (transduce
+   (comp
+    (remove (fn [[[ay need-setting] population]]
+              (= need-setting constants/non-send))) ;; remove non-send
+    ;; convert need-setting population to need-setting cost
+    (map (fn [[[ay need-setting] population]] [need-setting population]))
+    (map (fn [[need-setting population]] [(states/split-need-setting need-setting) population]))
+    (map (fn [[need-setting population]] [need-setting
+                                          (* (get cost-lookup need-setting 0)
+                                             population)]))
+    (map (fn [[need-setting cost]] cost)))
+   (fn
+     ([total] total)
+     ([total cost] (update total :total-cost + cost)))
+   {:total-cost 0.0}
+   calendar-year))
+
+(defn roll-up-total-in-send-by-ay-group [calendar-year]
+  (transduce
+   (comp
+    (map (fn [[[ay need-setting] population]] [ay population]))
+    (map (fn [[ay population]] [(u/ay-groups ay) population])))
+   (fn
+     ([totals] totals)
+     ([totals [ay-group population]]
+      (update totals ay-group m/some+ population)))
+   {}
+   calendar-year))
+
+(defn histogram-summary-rf
```

Comment:
This looks like code from utils, do you not need to also remove it from there?

Reply:
I've removed the unused code from utils now